### PR TITLE
Junit4 assignment's commit

### PR DESCRIPTION
**Pull request by Rishika0101**
Please review the code and let me know the changes that are required.

### CODE REVIEW

File: JUnit4Testing/src/main/java/org/example/StringUtils.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package org.example;
+public class StringUtils {
+
+    // This function is to reverse a string.
+    public static String reverse(String string) {
```
Comment (anup0060) [inline JUnit4Testing/src/main/java/org/example/StringUtils.java]: variable name should not be chosen as keyword name . please change

File: JUnit4Testing/src/main/java/org/example/StringUtils.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            char temp = chars[index];
+            chars[index] = chars[ArrayOfStrings];
+            chars[ArrayOfStrings] = temp;
+            index++;
+            ArrayOfStrings--;
+        }
+        return new String(chars);
+    }
+    // This function checks whether a string is empty or not.
+    public static boolean isEmpty(String string) {
```
Comment (anup0060) [inline JUnit4Testing/src/main/java/org/example/StringUtils.java]: same as above

File: JUnit4Testing/src/main/java/org/example/StringUtils.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        return string == null || string.isEmpty();
+    }
+    // This function checks whether a string is Palindrome or not.
+    public Boolean isPalindrome(String stringToCheck)
+    {
+        if(stringToCheck == "" || stringToCheck == null)
```
Comment (anup0060) [inline JUnit4Testing/src/main/java/org/example/StringUtils.java]: use paranthesis, even if single line add paranthesis

File: JUnit4Testing/src/test/java/StringUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    StringUtils stringUtils;
+    @Before
+    public void setup()
+    {
+        stringUtils = new StringUtils();
+    }
+
+    // Testing:- reverse method with non-empty string.
+    @Test
+    public void testReversee_test1()
```
Comment (anup0060) [inline JUnit4Testing/src/test/java/StringUtilsTest.java]: _test1 --> wrong 

_should_reverse_given_string , we need to mention what are we testing

**Review by anup0060 — CHANGES_REQUESTED** [review]

**Comment by Rishika0101** [comment]
As per your comment, I worked on it. Kindly check it and let me know if any mistakes are done.

File: JUnit4Testing_final/src/test/java/StringOperationTest.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    // Testing:- isEmpty method with empty method.
+    @Test
+    public void testEmpty_with_empty_method() {
+        Boolean emptyString = stringOperation.isEmpty("");
+        assertEquals(true, emptyString);
+    }
+
+    // Testing:- isPalindrome method with palindrome string.
+    @Test
+    public void testIsPalindrome_with_valid_palindrome_string() {
```
Comment (anup0060) [inline JUnit4Testing_final/src/test/java/StringOperationTest.java]: naming convention - testIsPalindrome_with_valid_palindrome_string  ---> 

testIsPalindrome_givenValidInput_shouldReturnWhetherPalindrome

**Review by anup0060 — CHANGES_REQUESTED** [review]

**Comment by Rishika0101** [comment]
Worked on comments and made the changes, please review and let me know the mistakes.

**Review by anup0060 — APPROVED** [review]

**Pull request merged**